File: extract_examples/generic.py
#!/usr/bin/python

import logging
import os.path
import zipfile
import sys
import uuid 

from adobe.pdfservices.operation.auth.credentials import Credentials
from adobe.pdfservices.operation.exception.exceptions import ServiceApiException, ServiceUsageException, SdkException
from adobe.pdfservices.operation.pdfops.options.extractpdf.extract_pdf_options import ExtractPDFOptions
from adobe.pdfservices.operation.pdfops.options.extractpdf.extract_element_type import ExtractElementType
from adobe.pdfservices.operation.execution_context import ExecutionContext
from adobe.pdfservices.operation.io.file_ref import FileRef
from adobe.pdfservices.operation.pdfops.extract_pdf_operation import ExtractPDFOperation
from adobe.pdfservices.operation.pdfops.options.extractpdf.extract_renditions_element_type import ExtractRenditionsElementType

logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))

def main(input,output):
	print(f"Going to extract {input} to {output}")

	# get base path.
	base_path = os.path.dirname(os.path.abspath(__file__))
	zip_file = base_path + "/" + str(uuid.uuid1()) + ".zip"

	try:
		
		# Initial setup, create credentials instance.
		credentials = Credentials.service_account_credentials_builder()\
			.from_file(base_path + "/pdftools-api-credentials.json") \
			.build()

		#Create an ExecutionContext using credentials and create a new operation instance.
		execution_context = ExecutionContext.create(credentials)
		extract_pdf_operation = ExtractPDFOperation.create_new()

		#Set operation input from a source file.
		source = FileRef.create_from_local_file(input)
		extract_pdf_operation.set_input(source)

		# Build ExtractPDF options and set them into the operation
		extract_pdf_options: ExtractPDFOptions = ExtractPDFOptions.builder() \
			.with_elements_to_extract([ExtractElementType.TEXT, ExtractElementType.TABLES]) \
			.with_elements_to_extract_renditions([ExtractRenditionsElementType.TABLES,ExtractRenditionsElementType.FIGURES]) \
			.build()

		extract_pdf_operation.set_options(extract_pdf_options)

		#Execute the operation.
		result: FileRef = extract_pdf_operation.execute(execution_context)

		# Save the result to the specified location.
		result.save_as(zip_file)

		file_to_extract = "structuredData.json"

		with zipfile.ZipFile(zip_file) as z:
			z.extractall(output)

		os.remove(zip_file)
		print(f"Done extracting to {output}")

	except (ServiceApiException, ServiceUsageException, SdkException):
		logging.exception("Exception encountered while executing operation")



if len(sys.argv) <= 2:
	print("Usage: generic.py input.pdf outputdirectory")
	sys.exit()

if __name__ == "__main__":
	main(sys.argv[1], sys.argv[2])

File: extract_examples/generic.js
#!/usr/bin/env node

const PDFServicesSdk = require('@adobe/pdfservices-node-sdk');

const chalk = require('chalk');
const fs = require('fs');
const AdmZip = require('adm-zip');
const { get } = require('http');
const nanoid = require('nanoid').nanoid;

let inputPDF = process.argv[2];
let output = process.argv[3];
let getStylingInfo = process.argv[4];


if(!inputPDF || !output) {
	console.error(chalk.red('Syntax: generic.js <input pdf doc> <output location folder> <get styling info, default true>'));
	process.exit(1);
}

if(!fs.existsSync(inputPDF)) {
	console.error(chalk.red(`Can't find input file ${inputPDF}`));
	process.exit(1);
}

if(!fs.existsSync(output)) {
	fs.mkdirSync(output);
}

let outputStatus = fs.statSync(output);
if(!outputStatus.isDirectory()) {
	console.error(chalk.red(`Output directory ${output} does not exist. Please make it first.`));
	process.exit(1);
}

let getStyling = true;
if(getStylingInfo === 'false') getStyling = false;

const credentials = PDFServicesSdk.Credentials
		.serviceAccountCredentialsBuilder()
		.fromFile('pdfservices-api-credentials.json')
		.build();

const config = PDFServicesSdk.ClientConfig.clientConfigBuilder().withConnectTimeout(360000).withReadTimeout(360000).build();

// Create an ExecutionContext using credentials
const executionContext = PDFServicesSdk.ExecutionContext.create(credentials,config);

// Build extractPDF options
const options = new PDFServicesSdk.ExtractPDF.options.ExtractPdfOptions.Builder()
			.addElementsToExtract(
				PDFServicesSdk.ExtractPDF.options.ExtractElementType.TEXT, 
				PDFServicesSdk.ExtractPDF.options.ExtractElementType.TABLES
			)
			.addElementsToExtractRenditions(
				PDFServicesSdk.ExtractPDF.options.ExtractRenditionsElementType.TABLES, 
				PDFServicesSdk.ExtractPDF.options.ExtractRenditionsElementType.FIGURES
			)
			.addTableStructureFormat(PDFServicesSdk.ExtractPDF.options.TableStructureType.CSV)
			.getStylingInfo(getStyling)
			.build()

// Create a new operation instance.
const extractPDFOperation = PDFServicesSdk.ExtractPDF.Operation.createNew(),
	input = PDFServicesSdk.FileRef.createFromLocalFile(
		inputPDF,
		PDFServicesSdk.ExtractPDF.SupportedSourceFormat.pdf
	);

extractPDFOperation.setInput(input);
extractPDFOperation.setOptions(options);

let outputZip = './' + nanoid() + '.zip';
extractPDFOperation.execute(executionContext)
	.then(result => result.saveAsFile(outputZip))
	.then(async () => {
		let zip = new AdmZip(outputZip);	
		await zip.extractAllTo(output);
		fs.unlinkSync(outputZip);
		console.log(chalk.green(`Extracted data from ${inputPDF} to output directory ${output}.`));
	})
	.catch(err => {
		for(let key in err) {
			console.log(key, '=', err[key]);
			console.log('----------------------');
		}
		console.log(err)
	});